Aptoide/appcoins-wallet-android
# Feature/appc 1232 use already existent transaction to proceed with purchase

**What does this PR do?**

   This PR introduces the recycling of adyen transactions. If a transaction is already available for the same item, reference and gateway it should be used instead of failing(some use cases) or creating a new one.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] PaymentMethodsPresenter.kt
- [ ] AdyenBillingService.kt

**How should this be manually tested?**
1 - Without a preselected method buy an item in trivial drive and check if it works
2 - With CC as preselected method buy an item in trivial drive and check if it works
3 - With CC as preselected method click oh "More payment methods" , select CC again and buy and item. It should work and use the same UID

  Flow on how to test this or QA Tickets related to this use-case: [APPC-1232](https://aptoide.atlassian.net/browse/APPC-1232)

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-1232](https://aptoide.atlassian.net/browse/APPC-1232)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new libs, new keys, etc.)




**Code Review Checklist**
